docs/userguide/miscellaneous.rst

```diff
@@ -94,3 +94,66 @@ correctly when installed as a zipfile, correct any problems if you can, and
 then make an explicit declaration of ``True`` or ``False`` for the ``zip_safe``
 flag, so that it will not be necessary for ``bdist_egg`` to try to guess
 whether your project can work as a zipfile.
+
+
+.. _Controlling files in the distribution:
+
+Controlling files in the distribution
+-------------------------------------
+
+For the most common use cases, ``setuptools`` will automatically find out which
+files are necessary for distributing the package.
+This includes all :term:`pure Python modules <Pure Module>` in the
+``py_modules`` or ``packages`` configuration, and the C sources (but not C
+headers) listed as part of extensions when creating a :term:`Source
+Distribution (or "sdist")`.
+
+However, when building more complex packages (e.g. packages that include
+non-Python files, or that need to use custom C headers), you might find that
+not all files present in your project folder are included in package
+:term:`distribution archive <Distribution Package>`.
+
+In these situations you can use a ``setuptools``
+:ref:`plugin <Adding Support for Revision Control Systems>`,
+such as :pypi:`setuptools-scm` or :pypi:`setuptools-svn` to automatically
+include all files tracked by your Revision Control System into the ``sdist``.
+
+.. _Using MANIFEST.in:
+
+Alternatively, if you need finer control, you can add a ``MANIFEST.in`` file at
+the root of your project.
+This file contains instructions that tell ``setuptools`` which files exactly
+should be part of the ``sdist`` (or not).
+A comprehensive guide to ``MANIFEST.in`` syntax is available at the
+:doc:`PyPA's Packaging User Guide <PyPUG:guides/using-manifest-in>`.
+
+Once the correct files are present in the ``sdist``, they can then be used by
+binary extensions during the build process, or included in the final
+:term:`wheel <Wheel>` [#build-process]_ if you configure ``setuptools`` with
+``include_package_data=True``.
+
+.. important::
+   Please note that, when using ``include_package_data=True``, only files **inside
+   the package directory** are included in the final ``wheel``, by default.
+
+   So for example, if you create a :term:`Python project <Project>` that uses
```

Comment:
How about adding something like: "If you are using `MANIFEST.in`, and you include a ``tests`` directory that is not included in the package director(ies), the ``tests`` will also be present in the ``sdist`` but not in the ``wheel``.   If using ``find_packages``, ensure that your glob pattern does not include the ``tests`` directory, or you use explicit ``include`` or ``exclude`` config along with ``find_packages`` to ensure that the ``tests`` folder is not included.  Note: if you accidentally include a top level ``tests`` folder (or another top level folder like ``docs``) in your ``packages`` config, it will end up in ``site-packages`` and get merged with any other package that accidentally does the same."

Reply:
However, I'm starting to think that all of this content should be in PyPUG and we should reference it from here so there is a single source of truth.

Reply:
Thank you very much @blink1073 for the text review.

I understand your concerns here, but I am starting to think that this is a non-trivial amount of information to be casually dropped in the middle of this section.

Maybe it would be best would be to add a separated section/page including the "most common configuration errors"?

That could live in the PyPUG, or even in setuptools docs if the maintainers find that too tool-specific.

Reply:
(I am already concerned about having 2 footnotes here... too much side-tracking)

Reply:
Good point about being tool-specific.  It seems like anything that isn't related to a PEP should go in these docs (same with flit-specific options for example).

Reply:
But perhaps that needs to wait until `setuptools` handles PEP 621 and PEP 660 so there is a uniform set of docs that can go in PyPUG and then refer out to tool-specific docs.

Reply:
I think I will stop the changes in this PR at this stage. Do you think it is in a good shape (although incomplete, I think it is an improvement over the current status of the docs...)